gtime/plotting/preprocessing.py

```diff
@@ -0,0 +1,199 @@
+import pandas as pd
+import numpy as np
+from scipy.linalg import toeplitz
+
+
+def seasonal_split(df: pd.DataFrame, cycle: str = "year", freq=None, agg="mean"):
```

Comment:
This function is way to long, create other functions to perform intermediate tasks

Reply:
fixed